### ISSUE TITLE
i have a problem

### ISSUE THREAD

**Issue by tttefan2005** [headshot2017/sm64-san-andreas#1]
this Project/Target: "plugin_sa - Release":
  The compiler's setup (GNU GCC Compiler) is invalid, so Code::Blocks cannot find/run the compiler.
  Probably the toolchain path within the compiler options is not setup correctly?!
  Do you have a compiler installed?
Goto "Settings->Compiler...->Global compiler settings->GNU GCC Compiler->Toolchain executables" and fix the compiler's setup.

Tried to run compiler executable 'C:\mingw32/bin/gcc.exe', but failed!

Skipping...
Nothing to be done (all items are up-to-date).

**Comment by headshot2017** [comment]
[https://github.com/headshot2017/sm64-san-andreas/wiki/Compiling-the-plugin#setting-up-codeblocks-and-mingw-w64](https://github.com/headshot2017/sm64-san-andreas/wiki/Compiling-the-plugin#setting-up-codeblocks-and-mingw-w64)
you need to make sure you downloaded mingw, extracted the "mingw32" folder to C:\, and set up the compiler settings as the wiki page shows you

**Comment by tttefan2005** [comment]
i did extract the folder to c but didnt work

**Comment by headshot2017** [comment]
and the compiler settings > toolchain executables? did you make sure the install directory is "C:\mingw32"? did you rename the compiler exes in it? like the C compiler must be "gcc.exe" instead of "mingw32-gcc.exe"? send a screenshot of the toolchain executables tab

**Comment by tttefan2005** [comment]
i did it now it works however heres another problem C:\plugin-sdk-master\shared\Other.h|105|error: no matching function for call to 'stat::stat(const char*&, stat*)'|

**Comment by headshot2017** [comment]
* Also [download this](https://cdn.discordapp.com/attachments/710557654992551936/1092008531047227422/shared.zip) zip file and extract inside the C:\plugin-sdk-master folder so that plugin-sdk can compile without errors on mingw-w64

you forgot this step

**Comment by tttefan2005** [comment]
heres another one 
||=== Build: GTASA Release in sm64-san-andreas (compiler: GNU GCC Compiler) ===|
ld.exe||cannot find -lplugin|
||error: ld returned 1 exit status|
||=== Build failed: 2 error(s), 0 warning(s) (0 minute(s), 5 second(s)) ===|

**Comment by headshot2017** [comment]
you have to finish building plugin_sa if you haven't, also, you need to double-check the build target you used; if you compiled plugin_sa with the "Debug" target, all the others must also be with the Debug target

**Comment by tttefan2005** [comment]
everytime when i open the game it freezes i think its beacsue of the libgcc_s_dw2-1.dll and libwinpthread-1.dll files

**Comment by headshot2017** [comment]
are you playing the steam version of the game?

**Comment by tttefan2005** [comment]
Actually i was playing the non steam original version of the game

**Comment by headshot2017** [comment]
if you go into your gta sa scripts folder do you see two sm64 .asi files?

**Comment by tttefan2005** [comment]
I only have one mario 64 así file

**Comment by headshot2017** [comment]
go to [this site](https://gtasnp.com/), upload your gta sa save file (My Documents > GTA San Andreas User Files > "GTASAsf3.b") and show me what the "Binary/EXE Version" says

**Comment by tttefan2005** [comment]
i already have save files

**Comment by headshot2017** [comment]
then upload your save file and show me the binary/exe version through the link i posted above

**Comment by tttefan2005** [comment]
i uploaded it and heres the screenshot
![bandicam 2023-04-06 20-05-19-880](https://user-images.githubusercontent.com/129924368/230526892-f341968d-e378-446a-b050-b12d25f7222f.jpg)

**Comment by headshot2017** [comment]
ok so it's v1.0, what if you delete every .dll and .asi file in the scripts folder, then build plugin_sa, libsm64 and sm64-san-andreas, all three with "Debug" target then try starting it again?

**Comment by tttefan2005** [comment]
i tried with debug but still dosent work

**Comment by tttefan2005** [comment]
it still freezes

**Comment by headshot2017** [comment]
ok what if you delete the asi and dll files again then try with Release on all of them? kinda grasping at straws here

**Comment by tttefan2005** [comment]
Im afriad it will casud the same freeze

**Comment by tttefan2005** [comment]
Now it works only i have to do is rename the rom file to sm64.us.z64

**Comment by headshot2017** [comment]
yes, that's all, and once you start the game you should hear the sm64 star sound
